### PR TITLE
Issue 5978

### PR DESCRIPTION
**Pull request by mpoiriert**
### PR checklist

- [ ] Read the [contribution guidelines](https://github.com/swagger-api/swagger-codegen/blob/master/CONTRIBUTING.md).
- [ ] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [ ] Filed the PR against the correct branch: `3.0.0` branch for changes related to OpenAPI spec 3.0. Default: `master`.
- [ ] Copied the [technical committee](https://github.com/swagger-api/swagger-codegen/#swagger-codegen-technical-committee) to review the pull request if your PR is targeting a particular programming langauge.

### Description of the PR

This is a pull request that contain the change made by this pullrequest

https://github.com/swagger-api/swagger-codegen/pull/6431

And the fix related to the conversation.

Let me know if there is something else.

### CODE REVIEW

**Comment by wing328** [comment]
Can we add tests to the JS ES6 client similar to what we've done in https://github.com/swagger-api/swagger-codegen/tree/master/samples/client/petstore/javascript/test ?

**Comment by wing328** [comment]
cc @CodeNinjai @frol @cliffano

File: samples/client/petstore/javascript/src/model/EnumArrays.js [hunk 1]
```diff
@@ -93,15 +93,15 @@
      * value: ">="
      * @const
      */
-    "GREATER_THAN_OR_EQUAL_TO": ">=",
+    "": ">=",
```
Comment (frol) [inline samples/client/petstore/javascript/src/model/EnumArrays.js]: This doesn't look right to me, does it?

File: samples/client/petstore/javascript/src/model/EnumArrays.js [hunk 2]
```diff
@@ -93,15 +93,15 @@
...
     /**
      * value: "$"
      * @const
      */
-    "DOLLAR": "$"  };
+    "value": "$"  };
```
Comment (frol) [inline samples/client/petstore/javascript/src/model/EnumArrays.js]: This doesn't look right to me, does it?

File: modules/swagger-codegen/src/main/resources/Javascript/es6/api_test.mustache [hunk 3]
```diff
@@ -1,15 +1,9 @@
...
-    // AMD.
-    define(['expect.js', '../../src/index'], factory);
-  } else if (typeof module === 'object' && module.exports) {
-    // CommonJS-like environments that support module.exports, like Node.
-    factory(require('expect.js'), require('../../src/index'));
-  } else {
-    // Browser globals (root is window)
-    factory(root.expect, root.{{moduleName}});
-  }
+  factory(expect, {{moduleName}});
```
Comment (frol) [inline modules/swagger-codegen/src/main/resources/Javascript/es6/api_test.mustache]: Does it still need to be ES5 style module (I mean the whole `(function(root, ...` seems to be redundant, but I am not 100% sure)?

File: samples/client/petstore/javascript-es6/docs/AdditionalPropertiesClass.md [hunk 4]
```diff
@@ -3,7 +3,7 @@
 ## Properties
 Name | Type | Description | Notes
 ------------ | ------------- | ------------- | -------------
-**mapProperty** | **{String: String}** |  | [optional] 
-**mapOfMapProperty** | **{String: {String: String}}** |  | [optional] 
+**map_property** | **{String: String}** |  | [optional]
```
Comment (frol) [inline samples/client/petstore/javascript-es6/docs/AdditionalPropertiesClass.md]: Should they stay in camelCase?

File: samples/client/petstore/javascript-es6/docs/ArrayOfArrayOfNumberOnly.md [hunk 5]
```diff
@@ -3,6 +3,6 @@
 ## Properties
 Name | Type | Description | Notes
 ------------ | ------------- | ------------- | -------------
-**arrayArrayNumber** | **[[Number]]** |  | [optional] 
+**ArrayArrayNumber** | **[[Number]]** |  | [optional]
```
Comment (frol) [inline samples/client/petstore/javascript-es6/docs/ArrayOfArrayOfNumberOnly.md]: Please, take a closer look if this really should be capitalized.

File: samples/client/petstore/javascript-es6/docs/FakeApi.md [hunk 6]
```diff
@@ -245,7 +245,7 @@ No authorization required
 
 <a name="testEndpointParameters"></a>
 # **testEndpointParameters**
-> testEndpointParameters(_number, _double, patternWithoutDelimiter, _byte, opts)
+> testEndpointParameters(_number, _double, pattern_without_delimiter, _byte, opts)
```
Comment (frol) [inline samples/client/petstore/javascript-es6/docs/FakeApi.md]: I think `pattern_without_delimiter` should be in camelCase.

File: samples/client/petstore/javascript-es6/docs/FakeApi.md [hunk 7]
```diff
@@ -340,14 +340,14 @@ import SwaggerPetstore from 'swagger_petstore';
...
 let opts = { 
-  'enumFormStringArray': ["enumFormStringArray_example"], // [String] | Form parameter enum test (string array)
-  'enumFormString': "-efg", // String | Form parameter enum test (string)
-  'enumHeaderStringArray': ["enumHeaderStringArray_example"], // [String] | Header parameter enum test (string array)
-  'enumHeaderString': "-efg", // String | Header parameter enum test (string)
-  'enumQueryStringArray': ["enumQueryStringArray_example"], // [String] | Query parameter enum test (string array)
-  'enumQueryString': "-efg", // String | Query parameter enum test (string)
-  'enumQueryInteger': 56, // Number | Query parameter enum test (double)
-  'enumQueryDouble': 1.2 // Number | Query parameter enum test (double)
+  'enum_form_string_array': ["enum_form_string_array_example"], // [String] | Form parameter enum test (string array)
```
Comment (frol) [inline samples/client/petstore/javascript-es6/docs/FakeApi.md]: camelCase?

**Comment by mpoiriert** [comment]
Hey guys, just a small touch back, I thought I have enough time to help on the project but I do not have enough to be efficient (starting a new job soon and will implicate myself more in the PHP Community down here...) I Might be able to make the change mention frol this weekend but it might be the last thing I would be able to do in a timely manner...

File: modules/swagger-codegen/src/main/resources/Javascript/es6/api_test.mustache (hunk 3 above)
Comment (mpoiriert) [reply modules/swagger-codegen/src/main/resources/Javascript/es6/api_test.mustache]: I am really not expert enough to be sure of that.

I can ask a friend to have a look if no one is sure...

**Comment by wing328** [comment]
@mpoiriert that's ok. We appreciate your help on this.

**Comment by kenisteward** [comment]
@CodeNinjai @frol @cliffano @wing328  I would like to help complete this.  What do you guys suggest be the process for me to do so?  Just take this branch from the other fork and go or what?

**Comment by frol** [comment]
@kenisteward Right, you can just pick up the commits and create another PR with extra changes.

**Comment by wing328** [comment]
@kenisteward picking up the commits like what @frol suggested or creating a new PR based on the latest master with your fixes are both fine with me.

**Pull request closed without merging**